internal/eval/eval.go

```diff
@@ -660,6 +656,79 @@ func evalJSONGET(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(string(resultBytes), false)
 }
 
+// evalJSONTOGGLE toggles a boolean value stored at the specified key and path.
+// args must contain at least the key and path (where the boolean is located).
+// If the key does not exist or is expired, it returns response.RespNIL.
+// If the field at the specified path is not a boolean, it returns an encoded error response.
+// If the boolean is `true`, it toggles to `false` (returns :0), and if `false`, it toggles to `true` (returns :1).
+// Returns an encoded error response if the incorrect number of arguments is provided.
+func evalJSONTOGGLE(args []string, store *dstore.Store) []byte {
+    if len(args) < 2 {
+        return diceerrors.NewErrArity("JSON.TOGGLE")
+    }
+
+    key := args[0]
+    path := args[1]
+
+    // Retrieve the object from the database
+    obj := store.Get(key)
+    if obj == nil {
+        return []byte("-ERR could not perform this operation on a key that doesn't exist\r\n")
+    }
+
+    // Check if the object is of JSON type
+    errWithMessage := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
+    if errWithMessage != nil {
+        return errWithMessage
+    }
+
+    jsonData := obj.Value
+
+    // Parse the JSONPath expression
+    expr, err := jp.ParseString(path)
+    if err != nil {
+        return diceerrors.NewErrWithMessage("invalid JSONPath")
+    }
+
+    // Execute the JSONPath query
+    results := expr.Get(jsonData)
+    if len(results) == 0 {
+        return clientio.Encode([]interface{}{}, false)  // Return empty array if no matches
+    }
+
+	toggleResults := make([]interface{}, len(results))
+
+    for i, result := range results {
+        switch v := result.(type) {
+        case bool:
+            newValue := !v
+            err := expr.Set(jsonData, newValue)
+            if err != nil {
+                return diceerrors.NewErrWithMessage("failed to set toggled value")
+            }
+
+            // Update the value in the store
+            newObj := &object.Obj{
+                Value:          jsonData,
+                TypeEncoding:   object.ObjTypeJSON,
+            }
+            store.Put(key, newObj)
```

Comment:
Hi, the changes look good. I have one question about a case where the previous object had an expired time set using expiry command.

Reply:
Yes, we need to ensure we set the expiration time on the new object too.